src/data-workspace/category-combo-table/use-value-matrix.js

```diff
@@ -0,0 +1,25 @@
+import { getIn } from 'final-form'
+import { useMemo } from 'react'
+import { useFormState } from 'react-final-form'
+/**
+ * Generates matrix (2d-array) of the values in a table, defined by dataElements and
+ * the categoryOptionCombos.
+ * @param {*} dataElements dataElements in order as rendered in table, these are "rows"
+ * @param {*} sortedCOCs categoryOptionCombos in order as rendered in table, these are the "columns"
+ */
+export const useValueMatrix = (dataElements = [], sortedCOCs = []) => {
+    const { values, active } = useFormState({
+        subscription: {
+            values: true,
+            active: true,
+        },
+    })
+
+    const matrix = useMemo(() => {
+        return dataElements.map((de) =>
+            sortedCOCs.map((coc) => getIn(values, `${de.id}.${coc.id}`))
+        )
+    }, [dataElements, sortedCOCs, active])
```

Comment:
If we would want this to update every time the value changes, we would switch out `active` with `values`. 
Currently this works so that it only updated when `active` is changed - and that is always changed `onBlur`.

Reply:
Please add this to the comment, otherwise it's not clear from reading the code